config/routes.rb

```diff
@@ -0,0 +1,15 @@
+Rails.application.routes.draw do
+
+  get '/login', to: "users#login_form", as: "login"
+  post '/login', to: "users#login"
+  post '/logout', to: "users#logout", as: "logout"
+
+  resources :works
+  post '/works/:id/upvote', to: 'works#upvote', as: "upvote"
+
+  resources :users, only: [:index, :show, :new]
+
+  get '/homepages', to: 'homepages#index', as: 'homepages'
```

Comment:
Since you have the root path going to homepages#index, why do you need this additional route?